### ISSUE TITLE
[BUG] [2.4.0] Failed to process project graph. Run "nx reset" to fix this. 

### ISSUE THREAD

**Issue by EelcoLos** [nx-dotnet/nx-dotnet#876]
### Current Behavior

After upgrading to `@nx-dotnet/core` to `2.4.0`, I get the following error when running any `nx` command

> Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.
>   An error occurred while processing files for the @nx-dotnet/core plugin.
>     - Plugin worker @nx-dotnet/core exited unexpectedly with code 0
>       Error: Plugin worker @nx-dotnet/core exited unexpectedly with code 0
>           at ChildProcess.<anonymous> (D:\DigitaalMeten\node_modules\nx\src\project-graph\plugins\isolation\plugin-pool.js:178:37)
>           at ChildProcess.emit (node:events:519:28)
>           at ChildProcess._handle.onexit (node:internal/child_process:294:12)
> The "@nx-dotnet/core" plugin threw an error while creating dependencies:
>     Error: Plugin worker timed out when processing message @nx-dotnet/core30304:createDependencies:1
>         at Timeout.<anonymous> (D:\DigitaalMeten\node_modules\nx\src\project-graph\plugins\isolation\plugin-pool.js:201:17)
>         at listOnTimeout (node:internal/timers:581:17)
>         at process.processTimers (node:internal/timers:519:7)

### Expected Behavior

to run nx commands properly

### Steps to Reproduce

1. update the `@nx-dotnet/core` package from 2.2.0 to 2.4.0
2. run `nx run-many -t build`
3. the error will show

### Nx Report

```shell
**NOTE**: had to reduce the `@nx-dotnet/core` package to produce this report
---

NX   Report complete - copy this into the issue template

Node   : 20.16.0
OS     : win32-x64
npm    : 10.8.2

nx (global)        : 19.5.0
nx                 : 19.5.3
@nx/js             : 19.5.3
@nx/jest           : 19.5.3
@nx/linter         : 19.5.3
@nx/eslint         : 19.5.3
@nx/workspace      : 19.5.3
@nx/angular        : 19.5.3
@nx/devkit         : 19.5.3
@nx/eslint-plugin  : 19.5.3
@nx/playwright     : 19.5.3
@nx/plugin         : 19.5.3
@nrwl/tao          : 19.5.3
@nx/web            : 19.5.3
@nx/webpack        : 19.5.3
typescript         : 5.5.4
---------------------------------------
Registered Plugins:
@nx/eslint/plugin
@nx-dotnet/core
@nx/jest/plugin
---------------------------------------
Community plugins:
@fortawesome/angular-fontawesome : 0.15.0
@ngrx/component                  : 18.0.1
@ngrx/component-store            : 18.0.1
@ngrx/effects                    : 18.0.1
@ngrx/entity                     : 18.0.1
@ngrx/router-store               : 18.0.1
@ngrx/schematics                 : 18.0.1
@ngrx/store                      : 18.0.1
@ngrx/store-devtools             : 18.0.1
@nx-dotnet/core                  : 2.4.0
---------------------------------------
```

### nx.json

```json
{
	"tasksRunnerOptions": {
		"default": {
			"runner": "nx-remotecache-azure",
			"options": {}
		}
	},
	"workspaceLayout": {
		"appsDir": "apps",
		"libsDir": "libs"
	},
	"plugins": [
		{
			"plugin": "@nx/eslint/plugin",
			"options": {},
			"inferredTargets": {
				"lint": "lint",
				"lint-fix": "lint-fix"
			}
		},
		{
			"plugin": "@nx-dotnet/core",
			"options": {
				"solutionFile": "Ibis.DigitaalMeten.sln",
				"inferProjects": true,
				"nugetPackages": {},
				"inferredTargets": {
					"build": {
						"inputs": [
							"production",
							"^production",
							"!{workspaceRoot}/packag*.*",
							"!{workspaceRoot}/**/packag*.*",
							"!{workspaceRoot}/package.json",
							"!{workspaceRoot}/package-lock.json"
						],
						"targetName": "build",
						"configurations": {
							"production": {
								"runtime": "linux-x64",
								"configuration": "Release",
								"noDependencies": false
							}
						}
					},
					"lint": "lint",
					"serve": "serve",
					"test": {
						"inputs": [
							"default",
							"^production",
							"!{workspaceRoot}/packag*.*",
							"!{workspaceRoot}/**/packag*.*",
							"!{workspaceRoot}/package.json",
							"!{workspaceRoot}/package-lock.json"
						],
						"targetName": "test",
						"cache": true,
						"configurations": {
							"ci": {
								"blame": false,
								"blameCrash": false,
								"logger": "trx"
							}
						}
					}
				},
				"ignorePaths": [],
				"tags": ["nx-dotnet"]
			}
		},
		{
			"plugin": "@nx/jest/plugin",
			"options": {
				"targetName": "test"
			}
		}
	],
	"generators": {
		"@nx/angular:application": {
			"style": "scss",
			"linter": "eslint",
			"unitTestRunner": "jest",
			"e2eTestRunner": "playwright"
		},
		"@nx/angular:library": {
			"linter": "eslint",
			"unitTestRunner": "jest",
			"changeDetection": "OnPush",
			"style": "scss",
			"standalone": true,
			"strict": true
		},
		"@nx/angular:component": {
			"style": "scss",
			"changeDetection": "OnPush"
		}
	},
	"$schema": "./node_modules/nx/schemas/nx-schema.json",
	"cli": {
		"cache": {
			"enabled": true
		}
	},
	"targetDefaults": {
		"build": {
			"dependsOn": ["^build"],
			"inputs": ["production", "^production"],
			"cache": true
		},
		"lint": {
			"inputs": [
				"default",
				"{workspaceRoot}/.eslintrc.json",
				"{workspaceRoot}/tools/eslint-rules/**/*"
			],
			"cache": true
		},
		"lint-fix": {
			"inputs": ["default", "{workspaceRoot}/.eslintrc.json"],
			"options": {
				"fix": true
			}
		},
		"serve": {
			"dependsOn": ["^build"]
		},
		"e2e": {
			"cache": true
		},
		"test": {
			"inputs": [
				"default",
				"^production",
				"{workspaceRoot}/jest.preset.js"
			],
			"executor": "@nx/jest:jest",
			"cache": true,
			"options": {
				"passWithNoTests": true,
				"jestConfig": "{projectRoot}/jest.config.ts"
			},
			"configurations": {
				"ci": {
					"ci": false,
					"coverageReporters": ["json"],
					"outputDirectory": "coverage/{projectRoot}",
					"coverageDirectory": "coverage/{projectRoot}"
				}
			}
		},
		"@nx/eslint:lint": {
			"cache": true,
			"inputs": [
				"default",
				"{workspaceRoot}/.eslintrc.json",
				"{workspaceRoot}/tools/eslint-rules/**/*"
			]
		}
	},
	"namedInputs": {
		"default": ["{projectRoot}/**/*", "sharedGlobals"],
		"sharedGlobals": [],
		"production": [
			"default",
			"!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)",
			"!{projectRoot}/tsconfig.spec.json",
			"!{projectRoot}/jest.config.[jt]s",
			"!{projectRoot}/.eslintrc.json",
			"!{projectRoot}/src/test-setup.[jt]s"
		]
	},
	"parallel": 6,
	"cacheDirectory": ".cache/nx",
	"defaultBase": "origin/master",
	"pluginsConfig": {
		"@nx/js": {
			"projectsAffectedByDependencyUpdates": "auto"
		}
	}
}
```

### Failure Logs

```shell
Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.
An error occurred while processing files for the @nx-dotnet/core plugin.
- Plugin worker @nx-dotnet/core exited unexpectedly with code 0
Error: Plugin worker @nx-dotnet/core exited unexpectedly with code 0
at ChildProcess. (D:\DigitaalMeten\node_modules\nx\src\project-graph\plugins\isolation\plugin-pool.js:178:37)
at ChildProcess.emit (node:events:519:28)
at ChildProcess._handle.onexit (node:internal/child_process:294:12)
The "@nx-dotnet/core" plugin threw an error while creating dependencies:
Error: Plugin worker timed out when processing message @nx-dotnet/core30304:createDependencies:1
at Timeout. (D:\DigitaalMeten\node_modules\nx\src\project-graph\plugins\isolation\plugin-pool.js:201:17)
at listOnTimeout (node:internal/timers:581:17)
at process.processTimers (node:internal/timers:519:7)
```

### Additional Information

I found that version `2.3.0` __DOES__ work, so the issue started at `2.3.1`

**Comment by AgentEnder** [comment]
Did it hang for 5 minutes before dying? The max message timeout should be 5 minutes.

Does it work if you run with `NX_ISOLATE_PLUGINS=false`? If not, what does the error look like

**Comment by EelcoLos** [comment]
> Did it hang for 5 minutes before dying? The max message timeout should be 5 minutes.
> 
> Does it work if you run with `NX_ISOLATE_PLUGINS=false`? If not, what does the error look like

yes, it was a while (could be 5 minutes) for the timeout to trigger.
With the env set I got:

```
The "@nx-dotnet/core" plugin threw an error while creating dependencies:
    TypeError: dotnetClient.getProjectReferences is not a function
        at Object.createDependencies (D:\DigitaalMeten\node_modules\@nx-dotnet\core\src\graph\create-dependencies.js:21:49)
        at LoadedNxPlugin.createDependencies (D:\DigitaalMeten\node_modules\nx\src\project-graph\plugins\internal-api.js:60:59)
        at D:\DigitaalMeten\node_modules\nx\src\project-graph\build-project-graph.js:226:47
        at Array.map (<anonymous>)
        at updateProjectGraphWithPlugins (D:\DigitaalMeten\node_modules\nx\src\project-graph\build-project-graph.js:223:47)
        at buildProjectGraphUsingContext (D:\DigitaalMeten\node_modules\nx\src\project-graph\build-project-graph.js:135:30)
        at async buildProjectGraphUsingProjectFileMap (D:\DigitaalMeten\node_modules\nx\src\project-graph\build-project-graph.js:86:24)
        at async buildProjectGraphAndSourceMapsWithoutDaemon (D:\DigitaalMeten\node_modules\nx\src\project-graph\project-graph.js:101:30)
        at async createProjectGraphAndSourceMapsAsync (D:\DigitaalMeten\node_modules\nx\src\project-graph\project-graph.js:208:25)
        at async createProjectGraphAsync (D:\DigitaalMeten\node_modules\nx\src\project-graph\project-graph.js:198:39)
```

**Comment by AgentEnder** [comment]
That's interesting, let me do some digging.

**Comment by AgentEnder** [comment]
Can you run `npm ls @nx-dotnet/dotnet`? That feels like its pulling a different version as we added the dotnet client get project references call in 2.3.1

**Comment by EelcoLos** [comment]
> npm ls @nx-dotnet/dotnet

i ran it:
```
digitaalmeten@19.5.3 D:\DigitaalMeten
└─┬ @nx-dotnet/core@2.4.0
  └── @nx-dotnet/dotnet@2.3.0
```

**Comment by AgentEnder** [comment]
Ah, it looks like this broke when I swapped from some custom deploy scripts to nx release. I'll get it fixed up.

In the meantime you should be able to rm -rf node modules + package lock to get latest versions

**Comment by AgentEnder** [comment]
This should be published now in 2.4.1, let me know if you still have issues

**Comment by EelcoLos** [comment]
can verify this works now
